File: be/pkg/server/adapter/schema/room.go
package schema

import (
	"github.com/google/uuid"
	"github.com/jostrzol/mess/pkg/server/core/room"
)

type Room struct {
	ID            uuid.UUID
	Players       int
	PlayersNeeded int
	IsStartable   bool
	IsStarted     bool
}

func RoomFromDomain(r *room.Room) *Room {
	return &Room{
		ID:            r.ID().UUID,
		Players:       len(r.Players()),
		PlayersNeeded: room.PlayersNeeded,
		IsStartable:   r.IsStartable(),
		IsStarted:     r.IsStarted(),
	}
}

File: fe/src/api/schema/room.ts
import {Room} from "@/model/room";
import {UUID} from "crypto";

export interface RoomDto {
  ID: UUID;
  Players: number;
  PlayersNeeded: number;
  IsStartable: boolean;
  IsStarted: boolean;
}

export const roomToModel = (room: RoomDto): Room => {
  return {
    id: room.ID,
    players: room.Players,
    playersNeeded: room.PlayersNeeded,
    isStartable: room.IsStartable,
    isStarted: room.IsStarted,
  };
};

